### PR TITLE
Domain improvements

### PR DESCRIPTION
**Pull request by Polymorphe57**

### CODE REVIEW

File: addons/web/static/src/core/domain.js [hunk 1]
```diff
@@ -291,6 +291,11 @@ function matchCondition(record, condition) {
 
     const fieldValue = typeof field === "number" ? field : record[field];
     switch (operator) {
+        case "=?":
```
Comment (aab-odoo) [inline addons/web/static/src/core/domain.js:294]: not sure to understand the point of this operator. Is there a task? Commit message should explain a bit

File: addons/web/static/src/core/domain.js [hunk 2]
```diff
@@ -135,8 +135,17 @@ export class Domain {
             /** @type {AST} */
             return new Domain(descr.toString());
         } else {
-            const rawAST = typeof descr === "string" ? parseExpr(descr) : toAST(descr);
-            this.ast = normalizeDomainAST(rawAST);
+            try {
+                const rawAST = typeof descr === "string" ? parseExpr(descr) : toAST(descr);
+                this.ast = normalizeDomainAST(rawAST);
```
Comment (aab-odoo) [inline addons/web/static/src/core/domain.js:140]: can this line crash with a random error? You don't need it in the try if it already throw the correct class of errors

File: addons/web/static/src/core/domain.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ import { toPyValue } from "./py_js/py_utils";
  * @typedef {DomainListRepr | string | Domain} DomainRepr
  */
 
-class InvalidDomainError extends Error {}
+export class InvalidDomainError extends Error {}
```
[the remark this answers is not in the document]
Comment (Polymorphe57) [reply addons/web/static/src/core/domain.js:14]: done

File: addons/web/static/src/core/domain.js (hunk 2 above)
Comment (Polymorphe57) [reply addons/web/static/src/core/domain.js]: Right. Good remark. Thank you

**Pull request closed without merging**